docs/sources/operations/autoscaling_queriers.md

```diff
@@ -0,0 +1,141 @@
+workload that varies throughout the day.
+To make Loki easier to operate and optimize the cost of running Loki at scale,
+we have designed a set of resources to help you autoscale your Loki queriers.
+
+## Prerequisites
+
+You need to run Loki in Kubernetes as a set of microservices. You need to use the query-scheduler.
+
+We recommend using [Kubernetes Event-Driven Autoscaling (KEDA)](https://keda.sh/) to configure autoscaling
+based on Prometheus metrics. Refer to [Deploying KEDA](https://keda.sh/docs/latest/deploy) to learn more
+about setting up KEDA in your Kubernetes cluster.
+
+## Scaling metric
+
+Because queriers pull queries from the query-scheduler queue and process them on the querier workers, you should scale metrics based on:
+
+- The scheduler queue size.
+- The queries running in the queriers.
+
+The query-scheduler exposes the `cortex_query_scheduler_inflight_requests` metric.
+It tracks the sum of queued queries plus the number of queries currently running in the querier workers.
+The following query is useful to scale queriers based on the inflight requests.
+
+```promql
+sum(
+  max_over_time(
+    cortex_query_scheduler_inflight_requests{namespace="loki-cluster", quantile="<Q>"}[<R>]
+  )
+)
+```
+
+Use the quantile (Q) and the range (R) parameters to fine-tune the metric.
+The higher Q is, the more sensitive the metric is to short-lasting spikes.
+As R increases, we reduce the variation over time in the metric.
+A higher R-value helps avoid the autoscaler from modifying the number of replicas too frequently.
+
+In our experience, we have found that a Q of 0.75 and an R of 2 minutes work well.
+You can adjust these values according to your workload.
+
+## Cluster capacity planning
+
+When configuring KEDA to scale your Loki queriers, we need to configure:
+
+- The threshold for scaling up and down
+- The scale down stabilization period
+- The minimum and the maximum number of queriers
+
+Querier workers process queries from the queue. You can configure each Loki querier to run several workers.
+To reserve workforce headroom to address workload spikes, our recommendation is not to use more than 75% of the workers.
+For example, if you configure the Loki queriers to run 6 workers, set a threshold of `floor(0.75 * 6) = 4`.
+
+Our recommendation for the minimum number of queriers is to run at least one querier, and to look at the average
+number of inflight requests 75% of the time in the last seven days, targeting a 75% utilization of the queriers.
+So if we use 6 workers per querier, we will use the following query:
+
+```promql
+clamp_min(ceil(
+    avg(
+        avg_over_time(cortex_query_scheduler_inflight_requests{namespace="loki-cluster", quantile="0.75"}[7d])
+    ) / scalar(floor(vector(6 * 0.75)))
+), 1)
+```
+
+Our recommendation for the maximum number of queriers is running the number of queriers that would be required
+to process all the inflight requests you had 50% of the time during the last seven days.
```

Comment:
The maximum number of queriers to run is equal to the number of queriers required to process 50% of the inflight requests during a seven-day timespan.

Reply:
This is not correct: "50% of the inflight requests" is not the same as "all the inflight requests you had 50% of the time". The latter refers to a quantile; the 50 percentile is the median but not half.

Reply:
Gotcha, ok what about "The maximum number of queriers to run is equal to the number of queriers required to process all inflight requests 50% of the time during a seven-day timespan."

does that work?

Reply:
That sounds perfect. Committed the changes. Thanks!